apps/server/src/modules/tool/common/uc/tool-permission-helper.ts

```diff
@@ -19,30 +24,32 @@ export class ToolPermissionHelper {
 		contextExternalTool: ContextExternalTool,
 		context: AuthorizationContext
 	): Promise<void> {
+		// loading of ressources should be part of the UC -> unnessasary awaits
+		const [authorizableUser, course]: [User, Course] = await Promise.all([
+			this.authorizationService.getUserWithPermissions(userId),
+			this.courseRepo.findById(contextExternalTool.contextRef.id),
+		]);
+
 		if (contextExternalTool.id) {
-			await this.authorizationService.checkPermissionByReferences(
-				userId,
-				AuthorizableReferenceType.ContextExternalToolEntity,
-				contextExternalTool.id,
-				context
-			);
+			this.authorizationService.checkPermission(authorizableUser, contextExternalTool, context);
 		}
 
-		await this.authorizationService.checkPermissionByReferences(
-			userId,
-			ContextTypeMapper.mapContextTypeToAllowedAuthorizationEntityType(contextExternalTool.contextRef.type),
-			contextExternalTool.contextRef.id,
-			context
-		);
+		// const type = ContextTypeMapper.mapContextTypeToAllowedAuthorizationEntityType(contextExternalTool.contextRef.type);
+		// no different types possible until it is fixed.
+		this.authorizationService.checkPermission(authorizableUser, course, context);
```

Comment:
If i implement a loader for you than it can be looked different like what you need. Because i do not know your needs.
How i should handle it. The method do you use should not called inside the server and this was documented, shared in sprint review and arc chapter with the first implementation of the authorisation service. 
The only other way is that you must do all changes by your self.